common/thorhelper/thorread.cpp

```diff
@@ -444,108 +602,414 @@ const void *BinaryDiskRowReader::nextRow()
             inputBuffer.read(eogPending);
         const byte * next = inputBuffer.queryRow();
 
-        if (likely(segMonitorsMatch(next))) // NOTE - keyed fields are checked pre-translation
+        if (likely(fieldFilterMatch(next))) // NOTE - keyed fields are checked pre-translation
+            return processor(sizeRead, next);
+    }
+}
+
+
+//Implementation of IAllocRowStream, return a row allocated with roxiemem
+const void *BinaryDiskRowReader::nextRow()
+{
+    return inlineNextRow([this](size32_t sizeRead, const byte * next){
```

Comment:
personally, I think more readable if lambda function defined on a separate line.

Reply:
Like this?
```
    return inlineNextRow(
        [this](size32_t sizeRead, const byte * next)
        {
            if (needToTranslate)
            {
                size32_t size = translator->translate(allocatedBuilder.ensureRow(), *this, next);
                return allocatedBuilder.finalizeRowClear(size);
            }
            else
            {
                size32_t allocatedSize;
                void * result = outputAllocator->createRow(sizeRead, allocatedSize);
                memcpy(result, next, sizeRead);
                return (const void *)outputAllocator->finalizeRow(sizeRead, result, allocatedSize);
            }
        }
    );

```

Reply:
I think that's clearer, but actually what I meant was to declare it with an auto.
So :
```
        auto processorFunc = [this](size32_t sizeRead, const byte * next)
        {
            if (needToTranslate)
            {
                size32_t size = translator->translate(allocatedBuilder.ensureRow(), *this, next);
                return allocatedBuilder.finalizeRowClear(size);
            }
            else
            {
                size32_t allocatedSize;
                void * result = outputAllocator->createRow(sizeRead, allocatedSize);
                memcpy(result, next, sizeRead);
                return (const void *)outputAllocator->finalizeRow(sizeRead, result, allocatedSize);
            }
        };

        return inlineNextRow(processorFunc);

```

Reply:
ok, I may change to that in a subsequent PR